tk_toolchain/cmd_line_tools/tk_build_qt_resources/__main__.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+####################################################################################
+
+import argparse
+import re
+import subprocess
+
+
+def process_import_line(module, import_text):
+    return (
+        f"from {import_text} import {module}\n"
+        f"for name, cls in {module}.__dict__.items():\n"
+        f"    if isinstance(cls, type): globals()[name] = cls\n"
+    )
+
+
+def verify_compiler(compiler):
+    try:
+        print("Verifying compiler: ", compiler)
+        version_compiler = subprocess.check_output(
+            [compiler, "--version"], text=True
+        ).strip()
+        return True, version_compiler
+    except Exception as error:
+        return False, error
+
+
+def build_qt(compiler, output, py_built_path, import_text):
+    print(f"The import text to replace will be '{import_text}'")
+    output_path = f"{py_built_path}/{output}.py"
+    subprocess.run(compiler.split(" "), stdout=open(output_path, "w"), check=True)
+    content = open(output_path, "r").read()
+    content = re.sub(
+        r"^from PySide2.QtWidgets(\s.*)?$", "", content, flags=re.MULTILINE
+    )
+    content = re.sub(r"^\s*$", "", content, flags=re.MULTILINE)
+    content = re.sub(
+        r"^from PySide2\.(\w+) import (.*)$",
+        lambda m: process_import_line(m.group(1), import_text),
+        content,
+        flags=re.MULTILINE,
+    )
+    content = content.replace("from PySide2 import", f"from {import_text} import")
+    open(output_path, "w").write(content)
+
+
+def build_ui(compiler, qt_ui_path, py_built_path, import_text, name):
+    build_qt(
+        f"{compiler} -g python --from-imports {qt_ui_path}/{name}.ui",
+        name,
+        py_built_path,
+        import_text,
+    )
+
+
+def build_res(compiler, qt_ui_path, py_built_path, import_text, name):
+    build_qt(
+        f"{compiler} -g python {qt_ui_path}/{name}.qrc",
```

Comment:
Instead of having both `build_ui` and `build_res` calling `build_qt`, could we instead have `build_ui` and `build_res` return "something" that the caller then pass to `build_qt`?